diff --git a/node_modules/react-quill/lib/index.js b/node_modules/react-quill/lib/index.js
index 25b2943..64619a9 100644
--- a/node_modules/react-quill/lib/index.js
+++ b/node_modules/react-quill/lib/index.js
@@ -350,7 +350,7 @@
     var editor = this.getEditor();
     var toolbar = this.getEditingArea().previousSibling;
     [].slice.call(toolbar.querySelectorAll('button')).forEach(function(button) {
-      button.setAttribute('tabindex', this.props.tabIndex || 0);
+      button.setAttribute('tabindex', this.props.tabIndex);
     }.bind(this));
   },
 
@@ -361,12 +361,12 @@
   },
 
   focus: function() {
-    var editor = ReactDOM.findDOMNode(this.editor);
+    var editor = this.editor.getEditor().root;
     editor.focus();
   },
 
   blur: function() {
-    var editor = ReactDOM.findDOMNode(this.editor);
+    var editor = this.editor.getEditor().root;
     editor.blur();
   },
 
diff --git a/node_modules/react-quill/lib/mixin.js b/node_modules/react-quill/lib/mixin.js
index 92433a7..6a51d9e 100644
--- a/node_modules/react-quill/lib/mixin.js
+++ b/node_modules/react-quill/lib/mixin.js
@@ -10,7 +10,7 @@
 	},
 
 	getEditingArea: function() {
-		return ReactDOM.findDOMNode(this).querySelector('.ql-editor');
+		return this.editingArea;
 	},
 
 	getEditor: function() {
diff --git a/node_modules/react-quill/lib/toolbar.js b/node_modules/react-quill/lib/toolbar.js
index 098b8b3..135d55b 100644
--- a/node_modules/react-quill/lib/toolbar.js
+++ b/node_modules/react-quill/lib/toolbar.js
@@ -107,7 +107,7 @@
 	},
 
 	render: function() {
-		return DOM.div({
+		return React.createElement('div', {
 			key: this.props.key,
 			id: this.props.id,
 			className: 'quill-toolbar ' + (this.props.className||''),
